src/main/java/net/minecraftforge/resource/ResourceCacheManager.java

```diff
@@ -69,35 +89,39 @@
      * @param indexOffThreadConfig True to index the file-tree off-thread.
      * @param pathBuilder          The path builder to use.
      */
+    @Deprecated(since = "1.19.2", forRemoval = true)
     public ResourceCacheManager(final boolean supportsReloading, final ForgeConfigSpec.BooleanValue indexOffThreadConfig, final BiFunction<PackType, String, Path> pathBuilder)
     {
         this.supportsReloading = supportsReloading;
-        this.indexOffThreadConfigOption = indexOffThreadConfig;
+        this.indexOnThreadConfigurationKey = Iterators.getLast(indexOffThreadConfig.getPath().iterator());
         this.pathBuilder = pathBuilder;
     }
 
-    public boolean shouldIndexOffThread()
```

Comment:
Removal of this method is a binary breaking change, as previously the class had not been considered internal. It should get re-added, `@Deprecated(forRemoval)` and it would return the inversed value of `shouldIndexOnThread`

Reply:
fwiw this was not considered public api when it was reverted in 1.18. So the core team may be taking the same view here.

Reply:
That whole revert was a breaking change. This PR sounds like its fixing a bug, not making a revert.